File: cases/taint/go/deep-relay-chain-negative/deep_relay_chain_negative.go
package dataflowbench

func dfb_source() string { // DFB-SOURCE: deep-relay-chain-input
	return "tainted"
}

func dfb_sink(value string) {} // DFB-SINK: deep-relay-chain-sink

func relay6(value string) { // DFB-WITNESS: deep-relay-chain-hop-six
	dfb_sink(value)
}

func relay5(value string) {
	relay6(value)
}

func relay4(value string) {
	relay5(value)
}

func relay3(value string) {
	relay4(value)
}

func relay2(value string) {
	relay3(value)
}

func relay1(value string) { // DFB-WITNESS: deep-relay-chain-hop-one
	relay2(value)
}

func run() {
	tainted := dfb_source()
	_ = tainted
	relay1("clean")
}

File: cases/taint/typescript/deep-relay-chain-negative/DeepRelayChainNegative.ts
function dfb_source(): string { // DFB-SOURCE: deep-relay-chain-input
  return "tainted";
}

function dfb_sink(value: string): void {} // DFB-SINK: deep-relay-chain-sink

function relay6(value: string): void { // DFB-WITNESS: deep-relay-chain-hop-six
  dfb_sink(value);
}

function relay5(value: string): void {
  relay6(value);
}

function relay4(value: string): void {
  relay5(value);
}

function relay3(value: string): void {
  relay4(value);
}

function relay2(value: string): void {
  relay3(value);
}

function relay1(value: string): void { // DFB-WITNESS: deep-relay-chain-hop-one
  relay2(value);
}

function run(): void {
  const tainted: string = dfb_source();
  relay1("clean");
}
